pkg/shell/cockpit-cpu-status.js

```diff
@@ -52,28 +48,50 @@ PageCpuStatus.prototype = {
                     max: 100,
                     show: true,
                     ticks: 5,
-                    tickFormatter: function(v) { return v + "%"; }},
+                    tickFormatter: function(v) { return (v / 10) + "%"; }},
             xaxis: {show: true,
                     ticks: [[0.0*60, "5 min"],
                             [1.0*60, "4 min"],
                             [2.0*60, "3 min"],
                             [3.0*60, "2 min"],
                             [4.0*60, "1 min"]]},
+            legend: { show: true },
             x_rh_stack_graphs: true
         };
 
-        this.plot = shell.setup_complicated_plot("#cpu_status_graph",
-                                                   resmon,
-                                                   [{color: "#4daf4a"},
-                                                    {color: "#377eb8"},
-                                                    {color: "#ff7f00"},
-                                                    {color: "#e41a1c"}
-                                                   ],
-                                                   options);
+        var metrics = [
+            { name: "cpu.basic.iowait", derive: "rate" },
+            { name: "cpu.basic.system", derive: "rate" },
+            { name: "cpu.basic.user", derive: "rate" },
+            { name: "cpu.basic.nice", derive: "rate" },
+        ];
+
+        self.channel = cockpit.metrics(1000, {
+            source: "internal",
+            metrics: metrics,
+            cache: "cpu-status-rate"
+        });
+
+        /* The grid shows us the last five minutes */
+        self.grid = cockpit.grid(1000, -300, -0);
+        metrics.forEach(function(metric) {
+            self.grid.add(self.channel, [ metric.name ]);
+        });
+
```

Comment:
From here on, we rely on the fact that grid.add has added the rows in order to grid.rows.  Creating a explicit array is cleaner, I think:

```
var data = metrics.map(function (metric) { return self.grid.add(self.channel, [ metric.name ]); });
```

and then passing `data` to setup_complicated_plot.